libav.in.js

```diff
@@ -73,33 +73,66 @@
     }
     libav.target = target;
 
+    function getFiles(conf) {
+        // define a similar function, adapted to your bundler, probably you will ignore base
+        if (conf.ext === "wasm") {
+            return new URL(conf.base + "/libav-" + conf.ver + "-" + conf.config + conf.dbg + "." + conf.topts + ".wasm");
+        } else if (conf.ext === "import") {
+            // not supported by the standard implementation, since it is only required for bundler
+            // and it would trigger the bundler and fail.
+            return null;
+            // example what it should do, if it is implemented externally
+            // return import(conf.base + "/libav-" + conf.ver + "-" + conf.config + conf.dbg + "." + conf.topts + ".js")
+        } else {
+            return new URL(conf.base + "/libav-" + conf.ver + "-" + conf.config + conf.dbg + "." + conf.topts + ".js");
+        }
+    } 
+
     // Now start making our instance generating function
     libav.LibAV = function(opts) {
         opts = opts || {};
         var base = opts.base || libav.base;
-        var toImport = base + "/libav-@VER-@CONFIG@DBG." + target(opts) + ".js";
+        var topts = target(opts);
+        var getfiles = libav.getFiles || getFiles;
+        var url = getfiles( { ext: "js", base: base, ver: "@VER", config: "@VER", dbg: "@DBG", topts: topts });
+        (globalThis || window || self).LibAV.wasmurl = getfiles( { ext: "wasm", base, ver: "@VER", config: "@VER", dbg: "@DBG", topts });
         var ret;
 
         return Promise.all([]).then(function() {
             // Step one: Get LibAV loaded
             if (!libav.LibAVFactory) {
                 if (nodejs) {
                     // Node.js: Load LibAV now
-                    libav.LibAVFactory = require(toImport);
+                    libav.LibAVFactory = require(url);
 
                 } else if (typeof Worker !== "undefined" && !opts.noworker) {
                     // Worker: Nothing to load now
 
                 } else if (typeof importScripts !== "undefined") {
                     // Worker scope. Import it.
-                    importScripts(toImport);
-                    libav.LibAVFactory = LibAVFactory;
+                    if (this !== undefined) { // this tells us, if we inside a module
+                        importScripts(url);
+                    } else {
+                        var imported = getfiles( { ext: "import", base: base, ver: "@VER", config: "@VER", dbg: "@DBG", topts: topts });
+                        return imported
+                            .then(function(mod) {
+                                if (globalThis.LibAVFactoryAsync)
```

Comment:
Yes, LibAVFactoryAsync is used here. And I forgot to replace globalThis here...

Reply:
Ahhhhh, there we go. For some reason I expected to find it in the extern-post part. My mistake.

Reply:
It was the only way, how to get the info out, if the bundler had turned it into a module. But may be there is a better way...?